models/topics.model.js

```diff
@@ -14,5 +14,23 @@ function selectTopics(){
     })
 }
 
+function selectArticles(id=null){
+    let sqlQueryStr = `SELECT * FROM articles`
+    const queryVals = []
 
-module.exports = {selectTopics, selectEndpoints}
\ No newline at end of file
+    if(id){
+        sqlQueryStr += ` WHERE article_id = $1`
+        queryVals.push(id)
+    }
+
+    return db.query(sqlQueryStr,queryVals)
+    .then((result)=>{
+		if (result.rows.length === 0) {
+			return Promise.reject({ status: 404, msg: "not found" });
+		}
+        return result.rows
```

Comment:
Here we are returning an array, which isn't really what we want if we are returning just one thing (such as an article). Here we want to return just the object inside the array. Therefore you may look to do `return result.rows[0]`

Reply:
Ah okay, I was anticipating re-using the same app/controller/model for task 5.  Would that be okay?